core/txpool/legacypool/legacypool.go

```diff
@@ -295,15 +296,15 @@ func (pool *LegacyPool) Filter(tx *types.Transaction) bool {
 // head to allow balance / nonce checks. The transaction journal will be loaded
 // from disk and filtered based on the provided starting settings. The internal
 // goroutines will be spun up and the pool deemed operational afterwards.
-func (pool *LegacyPool) Init(gasTip *big.Int, head *types.Header, reserve txpool.AddressReserver) error {
+func (pool *LegacyPool) Init(gasTip uint64, head *types.Header, reserve txpool.AddressReserver) error {
```

Comment:
@axelKingsley here's another case of using a u64 to hold the gas amount

Reply:
I think this one should be fine,
* In the other (fixed) cases, the problem was the conversion from a `big.Int` to a `uint64` and then to a `uint256` all inline, `uint256.NewInt(balance.Uint64())`. By contrast, this is just a signature change to accept only `uint64`
* I checked all the callers to this Init function, and none are doing the type conversions which caused us trouble. Most of them are just using `1`, but the ones that use a variable are now doing *less* type conversion, not more.

The only way to have the same zeroed-out value problem would be if a *caller* to this function took a big int and tried to convert it to uint64 blindly before calling this function. Since that's not happening, I think we're all good.